packages/lesswrong/components/vulcan-core/ScrollToTop.tsx

```diff
@@ -9,7 +11,7 @@ export default function ScrollToTop() {
   const isNotFirstMountRef = useRef(false)
   
   // Stringify `query` to guarantee referential stability for the useEffect dependency
-  const queryAsStr = JSON.stringify(query);
+  const queryAsStr = isLWorAF && stringify(query);
```

Comment:
@s-cheng and I decided this was a better default. Changing this will cause some behaviour changes elsewhere (e.g. changing filters on allPosts now won't rescroll to the top), we couldn't find any that were bug-like but there might be a few that we come across later